common/persistence/sql/sqlplugin/postgresql/execution.go

```diff
@@ -264,7 +267,10 @@ func (pdb *db) WriteLockExecutions(
 	filter sqlplugin.ExecutionsFilter,
 ) (int64, int64, error) {
 	var executionVersion sqlplugin.ExecutionVersion
-	err := pdb.conn.GetContext(ctx,
+	if pdb.tx == nil {
+		panic("cannot acquire execution write lock without a transaction")
```

Comment:
not sure if panic will always to captured. does returning an error work here?

The sanity check also looks quite random 🤔

Reply:
Whoops, that was added for debugging. I wanted to make sure we never called that outside of a transaction (we didn't in our tests) in case that was related to our RDS failover bug